src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -73,12 +75,11 @@ private ReadOnlySpan<char> FullPath
             {
                 if (_fullPath.Length == 0)
                 {
-                    ReadOnlySpan<char> directory = Directory;
-                    directory.CopyTo(_pathBuffer);
-                    _pathBuffer[directory.Length] = Path.DirectorySeparatorChar;
-                    ReadOnlySpan<char> fileName = FileName;
-                    fileName.CopyTo(_pathBuffer.Slice(directory.Length + 1));
-                    _fullPath = _pathBuffer.Slice(0, directory.Length + 1 + fileName.Length);
+                    Debug.Assert(Directory.Length + FileName.Length < _pathBuffer.Length,
```

Comment:
Should this be  `Directory.Length + FileName.Length` <= _pathBuffer.Length`?

Reply:
We always add the separator.